accounts-db/src/accounts_db.rs

```diff
@@ -2532,11 +2532,28 @@ impl AccountsDb {
         is_startup: bool,
         timings: &mut CleanKeyTimings,
         epoch_schedule: &EpochSchedule,
+        old_storages_policy: OldStoragesPolicy,
     ) -> CleaningCandidates {
         let oldest_non_ancient_slot = self.get_oldest_non_ancient_slot(epoch_schedule);
         let mut dirty_store_processing_time = Measure::start("dirty_store_processing");
-        let max_slot_inclusive =
-            max_clean_root_inclusive.unwrap_or_else(|| self.accounts_index.max_root_inclusive());
+        let max_root_inclusive = self.accounts_index.max_root_inclusive();
+        let max_slot_inclusive = max_clean_root_inclusive.unwrap_or(max_root_inclusive);
+
+        if old_storages_policy == OldStoragesPolicy::Clean {
+            let slot_one_epoch_old =
+                max_root_inclusive.saturating_sub(epoch_schedule.slots_per_epoch);
+            // do nothing special for these 100 old storages that will likely get cleaned up shortly
+            let acceptable_straggler_slot_count = 100;
+            let old_slot_cutoff =
+                slot_one_epoch_old.saturating_sub(acceptable_straggler_slot_count);
+            let (old_storages, old_slots) = self.get_snapshot_storages(..old_slot_cutoff);
```

Comment:
i was originally worried about the cost of getting these snapshots, but that worry was about 3 years ago. I assume this isn't too expensive to run every clean? clean is only once per 100s or so. I imagine this is fine. Not sure how else you'd get this. I was piggy backing off the storages collected for hash of all accounts previously.